src/Icon/Icon.tsx

```diff
@@ -1,43 +1,56 @@
 import PropTypes from 'prop-types';
-import { css } from 'styled-components';
+import { css, DefaultTheme, StyledComponent } from 'styled-components';
 import { createComponent } from '../utils';
 
-const Icon = createComponent({
-  name: 'Icon',
-  tag: 'i',
-  props: ({ name }: any) => ({
-    className: Icon.getClassName(name),
-  }),
-  style: ({ theme, size, color, disabled, onClick }: any) => {
-    const colorFromTheme = theme.colors[color];
-    const resolvedColor = colorFromTheme || color;
+interface IconProps {
+  name: string;
+  size?: number;
+  color?: string;
+  onClick?: () => void;
+  disabled?: boolean;
+}
 
-    return css`
-      color: ${resolvedColor || 'inherit'};
-      font-size: ${size ? `${size}px` : 'inherit'};
+// Need to declare type and use Object.assign to provide typings because Icon is being accessed during initialization
+type IconType = StyledComponent<'i', DefaultTheme, IconProps, never> & { getClassName: (name: string) => string };
 
-      ${disabled &&
-        css`
-          pointer-events: none;
-          opacity: 0.65;
-        `};
+const Icon: IconType = Object.assign(
+  createComponent<IconProps, 'i'>({
+    name: 'Icon',
+    tag: 'i',
+    props: ({ name }) => ({
+      className: Icon.getClassName(name),
```

Comment:
Instead of the `Object.assign` workaround, can we inline the `className` generation logic? I don't think we use this static `getClassName ` function outside of this component so it seems safe to do so. 
```suggestion
      className: `mdi mdi-${name}`,
```

Reply:
@erikshestopal  Is this true for the rest of the components?

Reply:
@erikshestopal ignore question, doesnt seem like any other components use it 😄

Reply:
@erikshestopal Removed getClassName, glad to see it gone.